app/routines/newflow/educator_signup/activate_account.rb

```diff
@@ -0,0 +1,33 @@
+# Sets the passed-in user's `state` to `'activated'`
+# and pushes the user to salesforce as a new lead
+# If the user is already `activated`, then it does nothing.
+module Newflow
+  module EducatorSignup
+    class ActivateAccount
+      lev_routine
+      uses_routine CreateSalesforceLead
+
+      protected ###############
+
+      def exec(user:)
+        return if user.activated?
+
+        create_salesforce_lead_for(user)
+        user.update!(state: User::ACTIVATED)
+        SecurityLog.create!(
+          user: user,
+          event_type: :user_updated,
+          event_data: {
+            user_became_activated: 'true'
+          }
+        )
+      end
+
+      private ###############
+
+      def create_salesforce_lead_for(user)
+        CreateSalesforceLead.perform_later(user: user)
```

Comment:
what happens if the creation of the salesforce lead fails in the activation of an account?  how would this be detected & reconciled?

Reply:
The Lev gem (as magical as it is) takes care of those matters/questions. Most importantly, it wraps everything in a transaction and nothing gets committed unless all of it is able to be committed. Also the job is retried automatically and fails after a few attempts.